aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -699,6 +699,30 @@ Tensor sub_zerotensor(const Tensor& self, const Tensor& other, const Scalar& alp
   return maybe_add_maybe_sub(self, other, -alpha);
 }
 
+Tensor linalg_cross_zerotensor(
+  const Tensor& input,
+  const Tensor& other,
+  const int64_t dim)
+{
```

Comment:
Add a check to ensure that after broadcasting, input.size(dim) != 3 or other.size(dim) != 3

Reply:
not totally sure, but seems to be already checked in `TORCH_PRECOMPUTE_META_FUNC(linalg_cross)`, no?

```cpp
  TORCH_CHECK(input_broadcasted.size(dim) == 3, "dimension ", dimension, " does not have size 3");
```

you can double-check with this:

```py
import torch

a = torch.randn(4, 3)
b = torch.randn(4, 3)
# b = torch._efficientzerotensor(4, 3)
res = torch.linalg.cross(a, b)
print(res)

# a = torch.randn(1, 3)  # a is broadcast to match shape of b
a = torch._efficientzerotensor(1, 3)
res = torch.linalg.cross(a, b, dim=0)
print(res)

```

output:

```
RuntimeError: dimension 0 does not have size 3
```

Reply:
oh right okay so this gets checked because of this call below:
```
auto meta_out = at::redispatch::linalg_cross(
    c10::DispatchKeySet(at::DispatchKey::Meta),
    input.to(device),
    other.to(device),
    dim);
```

ok SGTM